Patch to ensure png-1.5 and higher can be used, taken from
https://lists.gnu.org/archive/html/bug-plotutils/2011-01/msg00000.html
Author: J. Sassmannshausen (Imperial College London/UK)
diff --git a/plotutils-2.6.orig/libplot/z_write.c b/plotutils-2.6/libplot/z_write.c
index d76a7cc..173d4a9 100644
--- a/plotutils-2.6.orig/libplot/z_write.c
+++ b/plotutils-2.6/libplot/z_write.c
@@ -164,7 +164,7 @@ _pl_z_maybe_output_image (S___(Plotter *_plotter))
     }
 
   /* cleanup after libpng errors (error handler does a longjmp) */
-  if (setjmp (png_ptr->jmpbuf))
+  if (setjmp (png_jmpbuf (png_ptr)))
     {
       png_destroy_write_struct (&png_ptr, (png_info **)NULL);
       return -1;
@@ -444,7 +444,11 @@ _our_error_fn_stdio (png_struct *png_ptr, const char *data)
 #endif
     }
 
+#if (PNG_LIBPNG_VER < 10500)
   longjmp (png_ptr->jmpbuf, 1);
+#else
+  png_longjmp (png_ptr, 1);
+#endif
 }
 
 static void 
@@ -515,7 +519,11 @@ _our_error_fn_stream (png_struct *png_ptr, const char *data)
 #endif
     }
 
+#if (PNG_LIBPNG_VER < 10500)
   longjmp (png_ptr->jmpbuf, 1);
+#else
+  png_longjmp (png_ptr, 1);
+#endif
 }
 
 static void 
